qiskit/chemistry/ground_state_calculation/mes_factories/vqe_uccsd_factory.py

```diff
@@ -0,0 +1,145 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The minimum eigensolver factory for ground state calculation algorithms."""
+
+from typing import Optional
+import numpy as np
+
+from qiskit.aqua import QuantumInstance
+from qiskit.aqua.algorithms import MinimumEigensolver, VQE
+from qiskit.aqua.operators import ExpectationBase
+from qiskit.aqua.components.optimizers import Optimizer
+from qiskit.chemistry.components.variational_forms import UCCSD
+from qiskit.chemistry.qubit_transformations import FermionicTransformation
+from qiskit.chemistry.components.initial_states import HartreeFock
+
+from .mes_factory import MESFactory
+
+
+class VQEUCCSDFactory(MESFactory):
+    """A factory to construct a VQE minimum eigensolver with UCCSD ansatz wavefunction."""
+
+    def __init__(self,
+                 quantum_instance: QuantumInstance,
+                 optimizer: Optional[Optimizer] = None,
+                 initial_point: Optional[np.ndarray] = None,
```

Comment:
We may want to add gradient here once the gradient PR is done.

Reply:
I would say that this goes more towards backlog for upcoming releases. I also do not see how gradients fit here @Cryoris maybe can see this.

Reply:
Yes gradients can and should be added in the backlog 👍

Reply:
VQE will take a gradient object once the Gradients PR is merged. It evidently does not have to be exposed here, since the user can roll their own factory, but Gradients is a big part of this release hence the comment.